docs/products/kafka/concepts/upgrade-procedure.rst

```diff
@@ -0,0 +1,253 @@
+Apache Kafka® upgrade procedure
+===============================
+
+One of the benefits of using a managed service like Aiven for Apache Kafka® is the automated upgrade procedure. 
+
+The upgrade procedure is executed during:
+    
+* maintenance updates
+* plan changes
+* cloud region migrations
+* manual node replacements performed by an Aiven operator
+    
+All the above operations involve creating new broker nodes to replace existing ones.
+
+Upgrade procedure steps
+---------------------------
+
+The following set of steps are executed during an upgrade procedure, the example shows a possible scenario for a 3-node Apache Kafka service.
+
+.. mermaid::
+
+    flowchart TD;
+        subgraph KafkaCluster
+            subgraph Node3
```

Comment:
I think it will read better  if Node 1 -> 3 goes from left to right. Also, AB, AC, BC goes from left to right (in short, just to move complete blocks in opposite direction)